### DISCUSSION TITLE
Support for basic git actions

### DISCUSSION THREAD

**Discussion by michaldivis** [gitbutlerapp/gitbutler#3592] [Q&A]
Hi, is there a plan to include support for basic git commands in GitButler?

I mean:
- view other local/remote branches and switch to them
- merge/rebase branches
- resolve conflicts
- cherry pick

**Comment by FaithfulDev** [comment]
You should see all local and remote branches in the branch view. Here is the GitButler repo with lots of other peoples branches, only existing remotely.

![image](https://github.com/gitbutlerapp/gitbutler/assets/16925901/f49dc7b9-b8ec-4012-87f9-df6ba5fa7447)

You can "apply" them to your local workspace. This is the equivalent of "switching" to them. Keep in mind, that GitButler uses a new kind of branching system, which operates separately from git itself.

**Comment by itxashancode** [comment]
I'd check the roadmap - basic branch switching and merge/rebase are usually core before anything else. If you're seeing remote branches but can't switch, it's likely the branch picker isn't wired to `git checkout` yet. For conflicts, GitButler tends to surface them in the UI but you'll still need to resolve in your editor and stage the result. Cherry-pick support is often added after merge/rebase is stable. If you want to test, run `git branch -a` to confirm remotes are fetched, then try `git switch <branch>` manually. If that works but Butler doesn't, it's a UI gap, not a Git one. You could also open an issue tagging "improvement" with the exact flow you expect - the maintainers usually prioritize based on that.
